src/components/Navigator/QuickNavigationModal.vue

```diff
@@ -9,180 +9,368 @@
           .concat(index < userInput.length - 1 ? `[^${char.toLowerCase()}]*?` : '')
       ), '');
     },
     debounceInput: debounce(function debounceInput(value) {
-      this.debouncedInput = value;
-    }, 500),
-    fuzzyMatch: ({ debouncedInput, symbols, processedInputRegex }) => (
-      symbols.map((symbol) => {
-        const match = processedInputRegex.exec(symbol.title.toLowerCase());
+      // Remove space-character
+      this.debouncedInput = value.replace(/\s/g, '');
+    }, 250),
+    endingPointHook() {
+      // Reset selected symbol to the first one of the list
+      this.focusedIndex = 0;
+    },
+    fuzzyMatch({ debouncedInput, symbols, processedInputRegex }) {
+      return symbols.map((symbol) => {
+        const match = processedInputRegex.exec(symbol.title);
         // Dismiss if symbol isn't matched
         if (!match) return false;
 
         const matchLength = match[0].length;
         const inputLength = debouncedInput.length;
         // Dismiss if match length is greater than 3x the input's length
         if (matchLength > inputLength * 3) return false;
-
         return ({
+          uid: symbol.uid,
           title: symbol.title,
           path: symbol.path,
+          parents: this.getParents(symbol.parent),
+          type: symbol.type,
           inputLengthDifference: symbol.title.length - inputLength,
           matchLength,
           matchLengthDifference: matchLength - inputLength,
           start: match.index,
           substring: match[0],
         });
-      }).filter(Boolean)
-    ),
+      }).filter(Boolean);
+    },
+    handleKeyEnter() {
+      if (
+        this.noResultsWereFound
+        || !this.userInput.length
+      ) return;
+      this.$router.push(this.filteredSymbols[this.focusedIndex].path);
+      this.closeQuickNavigationModal();
+    },
     orderSymbolsByPriority(matchingSymbols) {
       return matchingSymbols.sort((a, b) => {
         // Shortests symbol match title have preference over larger titles
         if (a.matchLengthDifference > b.matchLengthDifference) return 1;
         if (a.matchLengthDifference < b.matchLengthDifference) return -1;
-        // Shortests symbol title have preference over larger titles
-        if (a.inputLengthDifference > b.inputLengthDifference) return 1;
-        if (a.inputLengthDifference < b.inputLengthDifference) return -1;
         // Matches at the beginning of string have relevance over matches at the end
         if (a.start > b.start) return 1;
         if (a.start < b.start) return -1;
+        // Shortests symbol title have preference over larger titles
+        if (a.inputLengthDifference > b.inputLengthDifference) return 1;
+        if (a.inputLengthDifference < b.inputLengthDifference) return -1;
         return 0;
       });
     },
-  },
-  watch: {
-    userInput: 'debounceInput',
+    scrollIntoView() {
+      this.$refs.match[this.focusedIndex].scrollIntoView({
+        block: 'nearest',
+        inline: 'start',
```

Comment:
These dont work in safari I think, just so you are aware.

Reply:
I think that the MDN docs browser compatibility section for `Element.scrollIntoView()` is outdated (https://developer.mozilla.org/es/docs/Web/API/Element/scrollIntoView).  For me is working as expected in Safari 15.3, and there are multiple issues regarding this.
Ex:
https://github.com/mdn/browser-compat-data/issues/17542
https://github.com/mdn/browser-compat-data/issues/9055

Reply:
Alright then :)